tests/framework/clients/dynamic/dynamic.go

```diff
@@ -79,14 +99,22 @@ func (c *ResourceClient) Create(ctx context.Context, obj *unstructured.Unstructu
 		return nil, err
 	}
 
-	c.ts.RegisterCleanupFunc(func() error {
-		err := c.Delete(context.TODO(), unstructuredObj.GetName(), metav1.DeleteOptions{}, subresources...)
-		if errors.IsNotFound(err) {
-			return nil
-		}
+	if needsCleanup(obj) {
+		c.ts.RegisterCleanupFunc(func() error {
+			err := c.Delete(context.TODO(), unstructuredObj.GetName(), metav1.DeleteOptions{}, subresources...)
+			if errors.IsNotFound(err) {
+				return nil
+			}
 
-		return err
-	})
+			name := unstructuredObj.GetName()
+			if unstructuredObj.GetNamespace() != "" {
+				name = unstructuredObj.GetNamespace() + "/" + name
+			}
+			gvk := unstructuredObj.GetObjectKind().GroupVersionKind()
+
+			return fmt.Errorf("%v (%v): %w", name, gvk, err)
```

Comment:
Nit: I would recommend supplying a meaningful error prefix, something like 
```suggestion
			return fmt.Errorf("unable to delete %v (%v): %w", name, gvk, err)
```

I'm also thinking that the GVK should go first, but that's really just preference.